src/infra/flow-system-inmem/src/services/flow/flow_service_inmem.rs

```diff
@@ -662,7 +667,8 @@ impl FlowServiceInMemory {
         flow: &mut Flow,
         schedule_time: DateTime<Utc>,
     ) -> Result<TaskID, InternalError> {
-        let logical_plan = flow.make_task_logical_plan();
+        let logical_plan_opts = self.get_logical_plan_options(&flow.flow_key);
+        let logical_plan = flow.make_task_logical_plan(&logical_plan_opts);
```

Comment:
Maybe this could be a single step - take a flow key, configuration, and give a logical plan.
We can move logical plan creation into the service from the flow type if that helps.

I think if you do this, you can avoid having this additional struct `LogicalPlanOptions`, which seems to be useless.